src/core_plugins/i18n/server/i18n/i18n.js

```diff
@@ -0,0 +1,124 @@
+import path from 'path';
+import Promise from 'bluebird';
+import { readFile } from 'fs';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+let defaultLocale = 'en';
```

Comment:
I didn't realize this. Thanks for the "heads up".

From your description, I think the  `defaultLocale` is associated to the server instance and therefore the class instance. I have updated it to remove the `defaultLocale` out of the `i18n` module and into the `UiI18n` class. I also tidied up subsequent code. Is it correct then that plugins which need to `getTranslations()` and want to merge against default locale translations will need to get the locale from  the server instance that loaded the plugin?

WRT to `_build:verifyTranslations`, I am not sure if any of the calls to i18n needs to be wrapped? It is just asking for the locales registered and doing a get on translations. Then verifying them. 

The resetting of the `i18n` module in the unit tests is because translations are loaded by core and Kibana plugin at start-up. Is the build up and tear down a big issue?